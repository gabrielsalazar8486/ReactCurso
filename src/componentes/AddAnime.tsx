import React, { Dispatch, SetStateAction, useState } from "react"
import PropTypes from "prop-types"

type props={
  // handleApp:(nombreAnime:string)=>void
  handleApp:Dispatch<SetStateAction<string[]>>
}


const AddAnime = ({handleApp}:props) => {
  const [inputValue, setInputValue] = useState('')
  const handleInputChange = ({value}:{value:string})=>{
    setInputValue(value)
  }
  const handleSubmit = (e:React.FormEvent<HTMLFormElement>)=>{
    e.preventDefault()
    if (inputValue.trim().length > 0){
      handleApp(animes => [inputValue, ...animes])
      setInputValue("")
    }
  }
  return (
    <>
      <form onSubmit={(e)=>{ handleSubmit(e) }}>
        <input
          type='text'
          value={inputValue}
          onChange={(e)=>{ handleInputChange(e.target)}}
        />
      </form>
    </>
  )
}

AddAnime.propTypes = {
  handleApp: PropTypes.func.isRequired
}

export default AddAnime