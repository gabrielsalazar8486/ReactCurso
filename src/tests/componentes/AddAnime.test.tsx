import { shallow } from "enzyme"
import AddAnime from "../../componentes/AddAnime"
import { vitest } from "vitest"

describe('Pruebas en el componeste AddAnime',()=>{
  const handleApp = vitest.fn()
  let wrapper = shallow(<AddAnime handleApp={handleApp} />)

  beforeEach(()=>{
    vitest.clearAllMocks()
    wrapper = shallow(<AddAnime handleApp={handleApp} />)
  })

  test('Debe de mostrar el componete correctamente',()=>{
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe de cambiar la caja de texto',()=>{
    const input = wrapper.find('input')
    const value = 'HunterX'
    input.simulate('change', {target:{ value }})
    const p = wrapper.find('p')

    expect(p.text().trim()).toBe(value)
  })

  test('No debe de postear la informacion con submit',()=>{
    wrapper.find('form').simulate('submit', { preventDefault() {}})
    expect( handleApp ).not.toHaveBeenCalled()
  })
})