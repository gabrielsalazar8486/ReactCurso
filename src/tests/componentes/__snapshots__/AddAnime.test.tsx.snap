// Vitest Snapshot v1

exports[`Pruebas en el componeste AddAnime > Debe de mostrar el componete correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;

exports[`Pruebas en el componeste AddAnime > No debe de postear la informacion con submit 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p>
      HunterX
    </p>
    <input
      onChange={[Function]}
      type="text"
      value="HunterX"
    />
  </form>
</Fragment>
`;
